frontend/app/src/pages/home/join.tsx

```diff
@@ -1,11 +1,40 @@
+import { useState } from "react";
 import { Page } from "@/components";
 import { FindingMatchCard, SelectPreferencesCard } from "@/features/matching";
+import { io } from "socket.io-client";
+import { useNavigate } from "react-router-dom";
+import { Preferences } from "@/types/room";
 
 function JoinPage() {
+  const navigate = useNavigate();
+  const socket = io("http://localhost:8004");
+  const [waiting, setWaiting] = useState<boolean>(false);
+
+  const joinRoom = (preferences: Preferences) => {
+    setWaiting(true);
+    socket.emit(
+      "join",
+      preferences,
+      (response: { status: string; roomId: string }) => {
```

Comment:
i think this can be abstracted into a type too
- it would be helpful to use `zod` to validate that the response does indeed fit this shape as well instead of typecasting it